normits_demand/models/notem/production_models.py

```diff
@@ -243,60 +252,52 @@ def run(self,
-                pure_demand.to_pickle(self.export_paths.pure_demand[year])
-
-            if export_reports:
-                du.print_w_toggle(
-                    "Exporting pure demand reports to disk...",
-                    verbose=verbose
-                )
-
-                report_seg = nd.get_segmentation_level('notem_hb_productions_pure_report')
-                pure_demand_paths = self.report_paths.pure_demand
-                pure_demand.aggregate(report_seg).write_sector_reports(
-                    segment_totals_path=pure_demand_paths.segment_total[year],
-                    ca_sector_path=pure_demand_paths.ca_sector[year],
-                    ie_sector_path=pure_demand_paths.ie_sector[year],
-                )
+        # ## GENERATE PURE DEMAND ## #
+        self._logger.info("Loading the population data...")
+        pop_dvec = self._read_land_use_data(year, verbose)
+
+        self._logger.info("Applying trip rates...")
+        pure_demand = self._generate_productions(pop_dvec, verbose)
+
+        if export_pure_demand:
+            self._logger.info("Exporting pure demand to disk...")
+            pure_demand.to_pickle(self.export_paths.pure_demand[year])
+
+        if export_reports:
+            self._logger.info("Exporting pure demand reports to disk...")
+            report_seg = nd.get_segmentation_level('notem_hb_productions_pure_report')
+            pure_demand_paths = self.report_paths.pure_demand
+            pure_demand.aggregate(report_seg).write_sector_reports(
+                segment_totals_path=pure_demand_paths.segment_total[year],
+                ca_sector_path=pure_demand_paths.ca_sector[year],
+                ie_sector_path=pure_demand_paths.ie_sector[year],
+            )
 
             # ## SPLIT PURE DEMAND BY MODE AND TIME ## #
-            du.print_w_toggle("Splitting by mode and time...", verbose=verbose)
+            self._logger.info("Splitting by mode and time..." )
             fully_segmented = self._split_by_tp_and_mode(pure_demand)
 
             # ## PRODUCTIONS TOTAL CHECK ## #
             if not pure_demand.sum_is_close(fully_segmented):
-                warnings.warn(
+                msg=(
                     "The production totals before and after mode time split are not same.\n"
                     "Expected %f\n"
                     "Got %f"
                     % (pure_demand.sum(), fully_segmented.sum())
                 )
+                self._logger.debug(msg)
+                warnings.warn(msg)
 
             # Output productions before any aggregation
             if export_fully_segmented:
-                du.print_w_toggle(
-                    "Exporting fully segmented productions to disk...",
-                    verbose=verbose,
-                )
                 fully_segmented.to_pickle(self.export_paths.fully_segmented[year])
+                self._logger.info("Exporting fully segmented productions to disk...")
```

Comment:
Make sure the messages are staying in the same place they originally were - this has moved to after the `to_pickle()` call, which means this log isn't actually true here!